runtime/planificator.js

```diff
@@ -13,6 +13,78 @@ import {SuggestionComposer} from './suggestion-composer.js';
 
 let defaultTimeoutMs = 5000;
 
+const now = () => {
+  if (typeof performance == 'object') {
+    return performance.now();
+  }
+  let time = process.hrtime();
+  return time[0] * 1000 + time[1] / 1000000;
+};
+
+class ReplanQueue {
+  constructor(planificator, options) {
+    this._planificator = planificator;
+    this._options = options || {};
+    this._options.defaultReplanDelayMs = this._options.defaultReplanDelayMs || 3000;
+
+    this._changes = [];
+    this._replanTimer = null;
+  }
+  addChange() {
+    this._changes.push(now());
+    if (this.isReplanningScheduled()) {
+      this.postponeReplan();
+    } else {
+      this.scheduleReplan(this._options.defaultReplanDelayMs);
+    }
+  }
+  isReplanningScheduled() {
+    return Boolean(this._replanTimer);
+  }
+  scheduleReplan(intervalMs) {
+    this.cancelReplanIfScheduled();
+    this._replanTimer = setTimeout(() => this._planificator._requestPlanning(), intervalMs);
+  }
+  cancelReplanIfScheduled() {
```

Comment:
most of these are internal only, no? Please mark them as such by using an underscore.